### PR TITLE
Add create new project usecase

### PR DESCRIPTION
**Pull request by robinlacey**
Creates a new project using a flat file to store the data, this will change in the future

### CODE REVIEW

File: lib/homes_england/gateway/file_project.rb [hunk 1]
```diff
@@ -0,0 +1,64 @@
+require 'json'
+class HomesEngland::Gateway::FileProject
```
Comment (naliwajek) [inline lib/homes_england/gateway/file_project.rb]: Linter alert :trollface: Add empty line between requires and a class def 🥇

File: lib/homes_england/gateway/file_project.rb [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+      project_data = JSON.parse(data) unless data == ''
+    end
+
+    project_data
+
+  rescue Errno::ENOENT
+    []
+  end
+
+  def deep_symbolize_keys(obj)
```
Comment (naliwajek) [inline lib/homes_england/gateway/file_project.rb]: I can see this being useful soon in other gateways 👏

File: lib/homes_england/use_case/create_new_project.rb [hunk 3]
```diff
@@ -0,0 +1,17 @@
+class HomesEngland::UseCase::CreateNewProject
+
+  def initialize(project_gateway:)
+    @project_gateway = project_gateway
+
+  end
```
Comment (naliwajek) [inline lib/homes_england/use_case/create_new_project.rb]: Newline.

File: lib/homes_england/use_cases.rb [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+  def self.register(builder)
+    builder.define_use_case :project_gateway do
+      HomesEngland::Gateway::FileProject.new(
+        file_path: ENV['PROJECT_FILE_PATH']
+      )
+    end
+
+    builder.define_use_case :create_new_project do
+      HomesEngland::UseCase::CreateNewProject.new(
+        project_gateway: builder.get_use_case(:project_gateway)
```
Comment (naliwajek) [inline lib/homes_england/use_cases.rb]: Wait... you're getting gateway with a `get_use_case` method? 🤔

**Comment by naliwajek** [comment]
LGTM. Make sure to expand factory (or add standalone gateway factory) later.

File: lib/homes_england/use_cases.rb (hunk 4 above)
Comment (robinlacey) [reply lib/homes_england/use_cases.rb]: For now, this will change later :)

**Pull request merged**